cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -732,8 +732,18 @@ private V1ObjectMeta createDeployment(V1ObjectMeta metadata,
 
     V1Deployment deployment = buildDeployment(metadata, runtimeSpecs, runtimeConfigLocation);
 
-    deployment = appsApi.createNamespacedDeployment(programRuntimeNamespace, deployment, "true", null, null);
-    LOG.info("Created Deployment {} in Kubernetes", metadata.getName());
+    try {
+      deployment = appsApi.createNamespacedDeployment(programRuntimeNamespace, deployment, "true", null, null);
+      LOG.info("Created Deployment {} in Kubernetes", metadata.getName());
+    } catch (ApiException e) {
+      if (e.getCode() == HttpURLConnection.HTTP_CONFLICT) {
+        LOG.warn("The kubernetes deployment already exists : {}. Ignoring resubmission of the deployment.",
+                 e.getResponseBody());
+      } else {
+        LOG.error("Failed to create kubernetes deployment: {}", e.getResponseBody());
```

Comment:
Skip the logging if the exception is re thrown.

Reply:
+1, log and rethrow is generally a bad pattern, usually ends up in duplicate logging all over the stack. I'm guessing this was added because we're not actually catching ApiException and the response body gets lost? (really wish the k8s library put it in the exception message) If so, we should be adding a catch ApiException somewhere higher up in the call stack.

Reply:
Added a new exception type to wrap APIException. As far as I can see, the checked exceptions are then caught at the end of the start() function and propagated upwards as runtime exceptions; if it's cleaner to just create a runtime exception at that level instead of directly wrapping the ApiException, we can do that too.